upgrade golang from v1.11 to v1.12 and add librdkafka lib & send access log entry to kafka

hopkings2008 opened the pull request.

hopkings2008 commented:
@thesues @cxt90730 could you please help to review this PR? and this PR just only modifies the yig.docker, makefile, and prepare_env.sh. the purpose of this PR is to upgrade the golang from v1.11 to v1.12 in yig docker image and remove the sudo in some scripts.

cannium commented:
Have you considered https://github.com/Shopify/sarama, if I recall correctly, it doesn't require `librdkafka`, so is easier to get CI working.

thesues commented:
Due to hopkings's comments, Ubuntu and  Centos already have librdkafka package. So CI and deployment do not have much work to do.
More importantly,  he said "librdkafka" is heard to be more "STABLE" than sarama

cannium commented on messagebus/types/message.go:
```diff
@@ -0,0 +1,18 @@
+type Message struct {
+	Topic   string
+	Key     string
+	ErrChan chan error
```
I'm a little confused about the purpose of `ErrChan`, to make synchronized calls?

hopkings2008 replied:
Yes，if you want to wait for the message is sent to the broker, you should  set the ErrChan and wait for the response. And you can refer to messagebus/tests for the usage.

cannium replied:
I believe the most common use case would be asynchronous, say application layer code just fire a message and forget about it. So 1) we might need some kind of error handling, like to resend failed message inside `messagebus` lib, as you did in PR#103  2) the benchmark `BenchmarkSendOneMsg` tests about the synchronous case and could be considered as a lower-bound performance.

hopkings2008 replied:
For async usage, just leave the ErrChan to nil and don't wait for ErrChan to finish.
1) confluent-kafka-go already has retry mechanism for sending message. Currently, we set 5000ms timeout when sends a message, that means if the network has some problems, the message will be delivered failed after 5s.
2) Yes, for benchmark, we need to measure what time a single message needs to be delivered 'successfully'.

cannium replied:
Cool, thanks for the explanation!

hopkings2008 commented:
Thanks @cannium for reviewing this PR. We have a investigation before for the kafka golang lib. And we find that sarama in 2018 doesn't support consumer group and many people suggest to use sarama cluster if he wants to use sarama.

hopkings2008 commented:
According to the discussion before, we can move all the tests into yig container and run those tests into that container. I have changed the dockerfile for yig container and added the librdkafka lib into that container. After that, I will enable all the tests. Currently, the newly added tests are run in yig container manually.

hopkings2008 commented:
@cannium @thesues  three creators of kafka setup the confluent company. So, librdkafka is very stable and very popular.

cannium commented:
> @cannium @thesues three creators of kafka setup the confluent company. So, librdkafka is very stable and very popular.

I know about confluent and I have no problem with librdkafka, just curious.

hopkings2008 commented:
> > @cannium @thesues three creators of kafka setup the confluent company. So, librdkafka is very stable and very popular.
> 
> I know about confluent and I have no problem with librdkafka, just curious.

thanks a lot. please help to review this pr and welcome your suggestions.
